app/Domain/Timesheets/Services/Timesheets.php

```diff
@@ -452,4 +453,59 @@ public function getBookedHourTypes(): array
     {
         return $this->timesheetsRepo->kind;
     }
+
+    public function pollTimesheets(): array|false
+    {
+        return $this->timesheetsRepo->getAllAccountTimesheets();
+    }
+
+    public function pollForUpdatedTimesheets(): array|false
+    {
+        $timesheets = $this->timesheetsRepo->getAllAccountTimesheets();
+
+        foreach ($timesheets as $key => $timesheet) {
+            $timesheets[$key]['id'] = $timesheet['id'] . '-' . $timesheet['modified'];
+        }
+
+        return $timesheets;
+    }
+
+    /**
+     * Create a timesheet entry via Zapier.
+     *
+     * @param array $values {
+     *     @type int    $ticketId     The I?D of the ticket.
+     *     @type int?    $userId       Th?e ID of the user.
+     *     @type string $date         The date in 'YYYY-MM-DD' format.
+     *     @type string $kind         The type of work done (e.g., 'DEVELOPMENT').
+     *     @type int    $hours        The number of hours worked.
+     *     @type float?  $rate         The rate of pay.
+     *     @type string? $description  A description of the work done.
+     *     @type bool?   $paid         Whether the time has been paid.
+     * }
+     *
+     * @return array {
+     *     @type bool $success Whether the timesheet entry was created successfully.
+     * }
+     *
+     * @throws InvalidArgumentException if required parameters are missing.
+     */
+    public function createTimesheetViaZapierAction($values): array
```

Comment:
Please rename. Is this new method needed or can we use an existing one?

Reply:
lets use existing `logTime` I agree